src/states/node.rs

```diff
@@ -1527,6 +1536,39 @@ impl Node {
         }
     }
 
+    fn handle_node_identify(&mut self,
+                            public_id: &PublicId,
+                            peer_id: &PeerId,
+                            is_tunnel: bool,
+                            outbox: &mut EventBox) {
+        let (peer_found, peer_valid) = if let Some(peer) =
+            self.peer_mgr.get_peer_by_name(public_id.name()) {
+            (true, peer.valid())
+        } else {
+            // This can be our joining node sending its relocated name, before we're expecting it.
+            if let Some(joining_node_key) = self.peer_mgr.get_joining_node(peer_id) {
+                if joining_node_key == public_id.signing_public_key() {
+                    return;
+                }
+            }
+            (false, false)
+        };
+
+        if !peer_found {
+            // These are peers we do not have in our peer manager yet. Could be because of a
+            // disconnect and reconnect via tunnel. We thus add them to the peer manager and
+            // indicate invalid to wait for group approval before moving to RT
+            self.peer_mgr
+                .insert_pending_approval_node(*peer_id, *public_id, is_tunnel);
+            return;
+        }
+
+        self.peer_mgr.set_peer_id(public_id.name(), *peer_id);
```

Comment:
If the name doesn't exist in the peer manager yet, this does nothing and returns `false`, but the result is ignored. How is the association between that name and peer ID stored, so that when it does get accepted, the peer is added?